src/pages/Posts/PostList.tsx

```diff
@@ -0,0 +1,44 @@
+import {
+  IonCard,
+  IonCardSubtitle,
+  IonCardHeader,
+  IonCardTitle,
+  IonCardContent,
+} from '@ionic/react';
+import React from 'react';
+import Interweave from 'interweave';
+import PostResource from '../../resources/post';
+
+interface PostListProps {
+  readonly post: PostResource;
+}
+
+const PostList: React.FC<PostListProps> = ({ post }) => {
+  const createPreviewText: Function = (body: string) => {
+    const regex = RegExp('^(.*?)[.?!]', 'u');
+    const previewText: string[] = body.split(regex);
+
+    if (previewText === null) {
+      return '';
+    } else {
+      return previewText.slice(1, 2).toString();
+    }
+  };
```

Comment:
It should be possible to do this entirely in CSS using the [line-clamp](https://css-tricks.com/almanac/properties/l/line-clamp/) and [text-overflow](https://www.w3schools.com/cssref/css3_pr_text-overflow.asp) properties. May be cleaner.